components/organisms/accounts/statement/statement.vue

```diff
@@ -63,102 +63,120 @@
-    }
-
-    get availableBalance() {
-        if (this.accountsStore.accountState.account) {
-            return this.accountsStore.accountState.account.balances.availableBalance
-        } else {
+    props: {
+        filters: {
+            type: Object as PropType<GetManagedAccountStatementRequest>,
+            required: true,
+        },
+    },
+    setup(props) {
+        const route = useRoute()
+        const { accounts } = useStores(['accounts'])
+        const { account, downloadAsCSV } = useAccounts()
+        const { monthsFilter } = useFilters()
+        const { setFilters } = useRouterFilter()
+
+        const filteredStatement = computed(() => accounts?.filteredStatement)
+
+        const availableBalance = computed(() => {
+            if (account) {
+                return account.value?.balances.availableBalance
+            }
+
             return 0
-        }
-    }
+        })
 
-    get filteredStatementLength(): number {
-        if (this.filteredStatement) {
-            return Object.keys(this.filteredStatement).length
-        } else {
+        const filteredStatementLength: ComputedRef<number> = computed(() => {
+            if (filteredStatement.value) {
+                return Object.keys(filteredStatement.value).length
+            }
             return 0
-        }
-    }
+        })
 
-    get filterDate() {
-        return {
-            start: this.filters.fromTimestamp,
-            end: this.filters.toTimestamp,
-        }
-    }
-
-    get months() {
-        return this.monthsFilter(this.account!.creationTimestamp)
-    }
-
-    // TODO: Resolve to useBase Composable
-    // Move all luxon function calls to utils for reusability
-    formatDate(val) {
-        const dateTime = DateTime.fromJSDate(val)
-        if (dateTime.hasSame(DateTime.now(), 'year')) {
-            return dateTime.toFormat('d MMMM')
-        }
-        return dateTime.toFormat('d MMMM yyyy')
-    }
+        const filterDate = computed(() => {
+            return {
+                start: props.filters.fromTimestamp,
+                end: props.filters.toTimestamp,
+            }
+        })
+
+        const months = computed(() => {
+            if (!account.value) return []
 
-    filterMonthChange(val) {
-        this.setFilters({
-            fromTimestamp: val.start,
-            toTimestamp: val.end,
+            return monthsFilter(account.value.creationTimestamp)
         })
-    }
 
-    downloadStatement() {
-        const _routeQueries = dot.object(this.$route.query)
-        const _filters = _routeQueries.filters ? _routeQueries.filters : {}
+        // TODO: Resolve to useBase Composable
+        // Move all luxon function calls to utils for reusability
+        const formatDate = (val) => {
+            const dateTime = DateTime.fromJSDate(val)
+            if (dateTime.hasSame(DateTime.now(), 'year')) {
+                return dateTime.toFormat('d MMMM')
+            }
```

Comment:
can we open a ticket for this ?

Reply:
done [here](https://weavr-payments.atlassian.net/browse/WEB-205)